src/main/java/com/wojciechbarwinski/demo/legendary_warehouse/exceptions/InsufficientStockException.java

```diff
@@ -0,0 +1,18 @@
+package com.wojciechbarwinski.demo.legendary_warehouse.exceptions;
+
+import com.wojciechbarwinski.demo.legendary_warehouse.dtos.InsufficientStockDTO;
+import lombok.Getter;
+
+import java.util.List;
+
+@Getter
+public class InsufficientStockException extends ApplicationException {
+    private final List<InsufficientStockDTO> insufficientStock;
+    private final String message;
+
+    public InsufficientStockException(String message, List<InsufficientStockDTO> insufficientStock) {
+        this.message = message;
```

Comment:
Tutaj pytanie, czy na pewno chcemy dawać możliwość manipulacją message temu, kto programuje rzucenie tego wyjątku. Dość często w parametrze konstruktora nie dajesz message, tylko ustawiasz go na sztywno w body konstruktora. W ten sposób zapewniasz unifikację wiadomości dla danego typu wyjątku (tj. każdy rzucony wyjątek tego typu będzie miał podobny message)

Reply:
nigdy o tym nie myslałem ale w sumie to faktycznie ma sens. 
Nie trzeba sie zastanawiać jak treścią 
a jak 1 wyjątek raz zwracał by wiadomośc zxc a raz qwe to byłaby lipa